icu4c/source/common/unicode/brkiter.h

```diff
@@ -93,6 +95,34 @@ U_NAMESPACE_BEGIN
  * are available at http://www.unicode.org/reports/tr14/ and
  * http://www.unicode.org/reports/tr29/.
  * <p>
+ * A boundary logically occurs between two adjacent code points. The boundary
+ * analysis functions in this file return the index of the code unit directly
+ * after a boundary - so a value of 0 means "the beginning of the string" and a
+ * value of u_strlen(string) means "the end of the string".
+ * <p>
+ * A span of text between two adjacent boundaries is called a "segment."
+ * The boundary analysis functions in this file behave correctly regardless of
+ * whether the current position is set to a boundary location or into the
+ * middle of a segment. When the current position points to a boundary, the two
+ * segments that straddle that boundary are called the "previous segment" and
+ * the "current segment."
+ * <p>
+ * If the current position is set to the middle of a surrogate pair, the
+ * boundary analysis functions in this file will move the current position to
+ * the beginning of the surrogate pair before any other processing. If the
+ * current position then points into the middle of a segment, these routines
+ * will move to the beginning or end of that segment. Otherwise, the current
+ * position points to a segment boundary, and these routines will move to the
+ * beginning of the previous segment or the end of the current segment.
+ * <p>
+ * The boundary analysis functions in this file may scan an unbounded number of
+ * characters around the current position or even the returned position. (One
+ * example is a long string of flag emoji. In order to know whether the current
```

Comment:
" (One example is a long string of flag emoji. In order to know whether the current position is in the middle of a single flag or between adjacent flags, the implementation counts how many regional indicator symbols occur in the string before the current position, to determine if the count is odd or even.)"
How about
"For example, while examing a long string of flag emoji, in order to decide whether the current position is in the middle of a single flag or between adjacent flags, the implementation need to counts how many regional indicator symbols occur in the string before the current position, to determine if the count is an odd or even number."